### PR TITLE
Resolve #12, #13, #14 and #17

### LINKED ISSUE #12: Scrappers remove tile deconst markers
Ok I don't think they mine tiles player.surface.find_entities(area)  :smiley:
Articulating - Today at 7:02 AM
deconstructible-tile-proxy is an entity
but it works in strange and mysterious ways
do scrappers just .destroy()
?*
nexela - Today at 7:05 AM
and raise the died event
OK that would be a bug then :smiley:
and I can probably use that proxy to deconstruc tiles

right now it finds the proxy
and destroys it
but all that does is remove the deconstruction command

### PR DESCRIPTION
**Pull request by articulating**
Fixes tile deconstruction to correctly remove tiles (#12). Adds repair functionality to constructor nanobots and adds a repair pack to their recipe(#13). Enables automatic filling of module requests (#17) (except if it is a type "assembling-machine" and there is no recipe set, due to a bug being fixed in 0.15). Adds code to pickup up items on ground underneath ghost entites when constructing(#14).

### CODE REVIEW

File: control.lua [hunk 1]
```diff
@@ -198,25 +211,29 @@ local function destroy_marked_items(player, pos, nano_ammo, deconstructors)
...
         --Loop through the minable products and add the item(s) to the list
         if entity.prototype.mineable_properties and entity.prototype.mineable_properties.minable then
-          local products = entity.prototype.mineable_properties.products
-          if products then
-            for _, item in pairs(products) do
-              table.add_values(item_list, item.name, item.amount or math.random(item.amount_min, item.amount_max))
-            end
+          products = entity.prototype.mineable_properties.products
+        elseif entity.name == "deconstructible-tile-proxy" then
+          local tile = entity.surface.get_tile(entity.position.x, entity.position.y)
```
Comment (Nexela) [inline control.lua]: A position table should work here also

File: control.lua [hunk 2]
```diff
@@ -198,25 +211,29 @@ local function destroy_marked_items(player, pos, nano_ammo, deconstructors)
     if entity.to_be_deconstructed(player.force) and (nano_ammo.valid and nano_ammo.valid_for_read) and not table_find(global.queued, find_match, entity) then
       if deconstructors then
         local item_list = {}
-
         entity.surface.create_entity{name="nano-cloud-small-deconstructors", position=entity.position, force="neutral"}
         nano_ammo.drain_ammo(1)
 
         --Get all the damn items and clear the inventories
         for item_name, count in pairs(get_all_items_inside(entity)) do
```
Comment (Nexela) [inline control.lua]: My error, but since you are here. we can optimize this by wrapping it in an entity.has_items_inside() if check

**Comment by articulating** [comment]
New commit resolves #17 and #13, and also resolves your comments.

**Comment by articulating** [comment]
Latest commits resolve #14.

**Pull request merged**